File: 프로그래머스/level1/K번째 수.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];

        for(int i = 0;i < commands.length;i++){
            int []buf = Arrays.copyOfRange(array, commands[i][0]-1, commands[i][1]);
            Arrays.sort(buf);
            answer[i] = buf[commands[i][2]-1];
        }
        
        return answer;
    }
}

File: 프로그래머스/level1/K번째 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer, tmp;
    int a, b, c;
    
    for(int i = 0;i < commands.size();i++){
        a = commands[i][0]-1;
        b = commands[i][1]-1;
        c = commands[i][2]-1;
        for(int j = a; j <= b;j++)
            tmp.push_back(array[j]);
        sort(tmp.begin(), tmp.end());
        answer.push_back(tmp[c]);
        tmp.clear();
    }
    return answer;
}